packages/reports/src/routes/fetchReport.ts

```diff
@@ -26,13 +27,24 @@ class FetchReportRouteHandler {
     this.aggregator = createAggregator(Aggregator);
   }
 
-  fetchReport = async (req: Request, res: Response): Promise<void> => {
+  fetchReport = async (req: ReportsRequest, res: Response): Promise<void> => {
     const params = req.params as FetchReportParams;
     const query = req.query as FetchReportQuery;
     const report = await req.models.report.findOne({ code: params.reportCode });
     if (!report) {
       throw new Error(`No report found with code ${params.reportCode}`);
     }
+
+    const orgUnit = await req.models.entity.findOne({ code: query.organisationUnitCode });
+    if (!orgUnit) {
+      throw new Error(`No entity found with code ${query.organisationUnitCode}`);
+    }
+    const country = await orgUnit.country();
+    const permissionGroup = await req.models.permissionGroup.findById(report.permission_group_id);
+
+    if (!req.accessPolicy.allows(country.code, permissionGroup.name)) {
```

Comment:
```suggestion
    if (!req.accessPolicy.allows(entity.country_code, permissionGroup.name)) {
```
Save a fetch? I'm not reviewing this properly - just got curious before heading home!

Reply:
Ah good point!

Reply:
Needs to be addressed